docs/api-reference.md

```diff
@@ -1730,3 +1730,235 @@ Resolves a relative path to an absolute URL within the application.
+
+Use a command's or query's `middleware` option to specify middleware:
+
+#### resolve-client:
+
+```js
+client.query(
+  {
+    name: "MyReadModel",
+    resolver: "all"
+  },
+  {
+    middleware: {
+      response: [
+        // An array of middleware that runs on server response
+        createMyResponseMiddleware({
+          // Middleware options
+        }),
+        ...
+      ],
+      error: [
+        // An array of middleware that runs on server error
+        createMyErrorMiddleware({
+          // Middleware options
+        }),
+        ...
+      ]
+    }
+  },
+  (error, result) => {
+    ...
+  }
+})
+```
+
+#### resolve-react-hooks:
+
+```js
+const myQuery = useQuery(
+  {
+    name: 'MyReadModel',
+    resolver: 'all'
+  },
+  {
+    middleware: {
+      response: [
+        // An array of middleware that runs on server response
+        createMyResponseMiddleware({
+          // Middleware options
+        }),
+        ...
+      ]
+      error: [
+        // An array of middleware that runs on server error
+        createMyErrorMiddleware({
+          // Middleware options
+        }),
+        ...
+      ]
+    }
+  },
+  (error, result) => {
+    ...
+  }
+```
+
+If you use multiple middleware functions, they run in the order that they are specified in the options object.
+
+#### Available Middlewares
+
+This section lists request middleware included into the resolve-client package. The following middleware is available:
+
+| Name                | Description                                                 |
+| ------------------- | ----------------------------------------------------------- |
+| [parseResponse]()   | Deserializes the response data if it contains valid JSON.   |
+| [retryOnError]()    | Retries the request if the server responds with an error.   |
+| [waitForResponse]() | Validates the response and retries if the validation fails. |
+
+##### parseResponse
+
+Deserializes the response data if it contains valid JSON. If the data is not JSON, the original string is kept. Initialized by the `createParseResponseMiddleware` factory function.
+
+This middleware takes no options. You can add it to a request as shown below:
+
+```js
+import { createParseResponseMiddleware } from 'resolve-client'
+...
+
+const { data } = await client.query(
+  {
+    name: 'articles',
+    resolver: 'all'
+  },
+  {
+    middleware: {
+      response: [createParseResponseMiddleware()]
+    }
+  }
+)
+```
+
+##### retryOnError
+
+Retries the request if the server responds with an error. Initialized by the `createRetryOnErrorMiddleware` factory function.
+
+The `retryOnError` middleware takes the following options:
+
+| Option Name | Description                                                          |
+| ----------- | -------------------------------------------------------------------- |
+| attempts    | The number of retries on error.                                      |
```

Comment:
Why 'on' error?

Reply:
Changed to "The number of retries if the server responds with an error. "

Reply:
Ok